### PR TITLE
Update documentation for Theming to use sx props

### PR DESCRIPTION
**Pull request by smeng9**

### CODE REVIEW

File: docs/Theming.md [hunk 1]
```diff
@@ -9,28 +9,25 @@ Whether you need to adjust a CSS rule for a single component, or change the colo
 
 ## Overriding A Component Style
 
-Every react-admin component provides a `className` property, which is always applied to the root element.
+Every react-admin component inherits an `sx` property from MUI, which is always applied to the root element.
```
Comment (djhi) [inline docs/Theming.md:12]:
```suggestion
Every react-admin component exposes the `sx` property from MUI, which is always applied to the root element.
```

File: docs/Theming.md [hunk 2]
```diff
@@ -111,27 +104,24 @@ Sometimes you want the format to depend on the value. The following example show
 ```jsx
 import * as React from 'react';
 import { NumberField, List, Datagrid, TextField, EditButton } from 'react-admin';
-import { makeStyles } from '@mui/material/styles';
-import clsx from 'clsx';
 
-const useStyles = makeStyles({
+const ColoredNumberFieldClasses = {
```
Comment (djhi) [inline docs/Theming.md:108]:
```suggestion
const ColoredNumberFieldStyles = {
```

File: docs/Theming.md [hunk 3]
```diff
@@ -111,27 +104,24 @@ Sometimes you want the format to depend on the value. The following example show
...
-            {...props}
-        />
-    );
 };
 
+const ColoredNumberField = (props) => (
+    <NumberField
+        sx={{
+            ...(props.record[props.source] < 100 &&
+                ColoredNumberFieldClasses.small),
```
Comment (djhi) [inline docs/Theming.md:117]:
```suggestion
                ColoredNumberFieldStyles.small),
```

File: docs/Theming.md [hunk 4]
```diff
@@ -111,27 +104,24 @@ Sometimes you want the format to depend on the value. The following example show
...
+            ...(props.record[props.source] >= 100 &&
+                ColoredNumberFieldClasses.big),
```
Comment (djhi) [inline docs/Theming.md:119]:
```suggestion
                ColoredNumberFieldStyles.big),
```

File: docs/Theming.md [hunk 5]
```diff
@@ -154,26 +144,24 @@ Furthermore, you may extract this highlighting strategy into a Higher Order Comp
...
+const colored = (WrappedComponent) => (props) =>
+    (
         <WrappedComponent
-            className={clsx({
-                [classes.small]: props.record[props.source] < 500,
-                [classes.big]: props.record[props.source] >= 500,
-            })}
+            sx={{
+                ...(props.record[props.source] < 100 &&
+                    ColoredNumberFieldClasses.small),
```
Comment (djhi) [inline docs/Theming.md:158]:
```suggestion
                    ColoredNumberFieldStyles.small),
```

File: docs/Theming.md [hunk 6]
```diff
@@ -154,26 +144,24 @@ Furthermore, you may extract this highlighting strategy into a Higher Order Comp
...
+                ...(props.record[props.source] >= 100 &&
+                    ColoredNumberFieldClasses.big),
```
Comment (djhi) [inline docs/Theming.md:160]:
```suggestion
                    ColoredNumberFieldStyles.big),
```

**Review by djhi — CHANGES_REQUESTED** [review]
Thanks for taking the time to do this!

File: docs/Theming.md (hunk 1 above)
Comment (WiXSL) [reply docs/Theming.md:12]: The `className` information is still valid. I wouldn't replace one over the other, but add the `sx` prop to it

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Comment by panfiva** [comment]
Can we also add information how to customize styles for nested elements inside components, for example, this is how I customized `content` component inside `Layout`; however if there is a better way, it would be nice to know:

```js
const StyledLayout = styled(Layout)({
	'.RaLayout-content': {
		paddingBottom: 0,
		paddingLeft: 8,
		paddingRight: 8,
	},
})

const LayoutComponent: FC<CoreLayoutProps> = (props) => {
	return (
		<StyledLayout
			{...props}
			// menu={Menu}
			error={ErrorWithLogging}
		/>
	)
}
export default LayoutComponent
```

**Review by djhi — APPROVED** [review]
Thanks!

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**